embassy-stm32/src/crs.rs

```diff
@@ -0,0 +1,67 @@
+use embassy_hal_common::PeripheralRef;
+
+use crate::pac::CRS as PAC_CRS;
+use crate::peripherals::CRS;
+use crate::rcc::enable_hsi48;
+use crate::rcc::sealed::RccPeripheral;
+use crate::{pac, Peripheral};
+
+pub struct Config {
+    sync_source: SyncSource,
+}
+
+pub enum SyncSource {
+    Gpio,
+    Lse,
+    Usb,
+}
+
+pub struct Crs<'d> {
```

Comment:
I think CRS should be fields in the RCC `Config` struct, instead of a user-facing driver struct. It's kind of part of RCC, and as a user you don't do much with it directly other than configuring it at boot.

Reply:
Works for me, I went back and forth on how the config should work for this a few times. I'll get that changed.